jni-gen/builtins/utils.rs

```diff
@@ -0,0 +1,39 @@
+use jni::{errors::Result, objects::JObject, JNIEnv};
+use jni::strings::JavaStr;
+use std::ffi::CStr;
+
+fn java_str_to_string(string: &JavaStr) -> Result<String> {
+    unsafe { match CStr::from_ptr(string.get_raw()).to_str() {
+        Ok(resultstring) => Ok(resultstring.to_string()),
+        Err(_) => Ok("Cannot be determined".to_string()),
+    } }
+}
+
+fn get_jobject_class_name_inner<'a, 'c, O>(env: &'c JNIEnv<'c>, object: O) -> Result<String>
+where
+    O: Into<JObject<'a>>
+{
+    let object_class = env.get_object_class(object)?;
+    let res = java_str_to_string(
+        &env.get_string(
+            env.call_method(object_class, "getName", "()Ljava/lang/String;", &[])?
+                 .l()?
+                 .into(),
+        )?,
+    );
+
+    match res {
+        Ok(str) => Ok(str.replace(".", "/")),
+        Err(_) => Ok("Cannot be determined".to_string()),
+    }
+}
```

Comment:
Could you add a check of the type of the `object` parameter, executed only when compiling in debug mode?

Reply:
I thought that the argument had to be an instance of a `java.lang.Class` and I was asking for an `assert_debug!` subtyping check, but I now see that I was wrong because `get_object_class` takes care of retrieving the class. So, nothing to change here.